### PR TITLE
Replace deprecated e.keyCode

### PR DESCRIPTION
**Pull request by sharpirate**
## Summary
This PR refactors the checks for user key press to use [KeyboardEvent.key](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) instead of the deprecated [KeyboardEvent.keyCode](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode)

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions
The easiest way to test this is to test the Jetpack connection banner:

1. Open your jurassic tube testing site
2. Go to **/wp-admin/plugins.php**
3. Deactivate Jetpack
4. Activate Jetpack
5. You should see the following screen:

image

6. Press the **Escape** key. You should see the screen being dismissed.

### CODE REVIEW

**Review by obenland — DISMISSED** [review]
I tested the changes in `[jetpack-connection-banner.js](https://github.com/Automattic/jetpack/pull/25994/files#diff-ad266fea8e26467cd6ca2e98bfad645ef8698764ff68386c944052d99d84beea)` with a jurassic.ninja site and it's working well!

Not sure how to test the other instances, but based on code review I don't think they'll behave any differently.

File: projects/plugins/jetpack/_inc/client/components/jetpack-dialogue-modern/index.jsx [hunk 1]
```diff
@@ -7,7 +7,8 @@ import onKeyDownCallback from 'utils/onkeydown-callback';
 
 class ModernOverlay extends Component {
 	maybeDismiss = e => {
-		if ( this.props.showDismiss && ( ! e.keyCode || e.keyCode === 27 ) ) {
+		// "Esc" is an IE/Edge specific value
```
Comment (anomiex) [inline projects/plugins/jetpack/_inc/client/components/jetpack-dialogue-modern/index.jsx]: Our browserslist config indicates that we don't support IE anymore, and only Edge >= 102. Is "Esc" still relevant?

https://caniuse.com/keyboardevent-key lists Edge as having non-standard identifiers up to version 18, and full support after that.

**Review by anomiex — COMMENTED** [review]
Thanks for doing this!

I note you replaced it with `event.key` rather than `event.code`. Was there a reason for that choice? The latter seems closer to the old behavior, but maybe the old behavior wasn't actually what was intended.

Did you intend to do just one component? I see several more uses in the code.

<details><summary>List</summary>

* [projects/plugins/jetpack/_inc/accessible-focus.js line 8](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/_inc/accessible-focus.js#L8)
* [projects/plugins/jetpack/_inc/client/components/chart/bar.jsx line 81](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/_inc/client/components/chart/bar.jsx#L81)
* [projects/plugins/jetpack/_inc/client/components/popover/index.jsx line 158](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/_inc/client/components/popover/index.jsx#L158)
* [projects/plugins/jetpack/_inc/client/components/popover/menu.jsx line 114](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/_inc/client/components/popover/menu.jsx#L114)
* [projects/plugins/jetpack/_inc/client/components/section-nav/tabs.jsx line 149](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/_inc/client/components/section-nav/tabs.jsx#L149)
* [projects/plugins/jetpack/_inc/client/components/select-dropdown/index.jsx line 269](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/_inc/client/components/select-dropdown/index.jsx#L269)
* [projects/plugins/jetpack/_inc/client/lib/accessible-focus/index.js line 9](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/_inc/client/lib/accessible-focus/index.js#L9)
* [projects/plugins/jetpack/_inc/jetpack-deactivate-dialog.js lines 38–45](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/_inc/jetpack-deactivate-dialog.js#L38-L45)
* [projects/plugins/jetpack/extensions/blocks/calendly/calendly.php line 204](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/calendly/calendly.php#L204)
* [projects/plugins/jetpack/extensions/blocks/contact-form/edit.js line 284](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/contact-form/edit.js#L284)
* [projects/plugins/jetpack/extensions/blocks/donations/controls.js line 50](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/donations/controls.js#L50)
* [projects/plugins/jetpack/extensions/blocks/donations/view.js line 162](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/donations/view.js#L162)
* [projects/plugins/jetpack/extensions/blocks/eventbrite/eventbrite.php line 236](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/eventbrite/eventbrite.php#L236)
* [projects/plugins/jetpack/extensions/blocks/image-compare/view.js line 429](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/image-compare/view.js#L429)
* [projects/plugins/jetpack/extensions/blocks/image-compare/view.js line 451](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/image-compare/view.js#L451)
* [projects/plugins/jetpack/extensions/blocks/image-compare/view.js line 460](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/image-compare/view.js#L460)
* [projects/plugins/jetpack/extensions/blocks/map/lookup/index.js line 125](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/map/lookup/index.js#L125)
* [projects/plugins/jetpack/extensions/blocks/markdown/edit.js line 54](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/markdown/edit.js#L54)
* [projects/plugins/jetpack/extensions/blocks/send-a-message/whatsapp-button/configuration.js line 55](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/send-a-message/whatsapp-button/configuration.js#L55)
* [projects/plugins/jetpack/extensions/blocks/story/player/index.js line 49](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/story/player/index.js#L49)
* [projects/plugins/jetpack/extensions/blocks/story/player/modal/frame.js lines 20–33](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/story/player/modal/frame.js#L20-L33)
* [projects/plugins/jetpack/extensions/blocks/tiled-gallery/gallery-image/edit.js line 23](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/blocks/tiled-gallery/gallery-image/edit.js#L23)
* [projects/plugins/jetpack/extensions/shared/components/block-styles-selector/index.js line 96](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/shared/components/block-styles-selector/index.js#L96)
* [projects/plugins/jetpack/extensions/shared/external-media/media-browser/index.js lines 150–161](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/shared/external-media/media-browser/index.js#L150-L161)
* [projects/plugins/jetpack/extensions/shared/external-media/sources/with-media.js line 70](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/extensions/shared/external-media/sources/with-media.js#L70)
* [projects/plugins/jetpack/modules/sharedaddy/admin-sharing.js line 341](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/modules/sharedaddy/admin-sharing.js#L341)
* [projects/plugins/jetpack/modules/widgets/social-icons/social-icons-admin.js line 100](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/modules/widgets/social-icons/social-icons-admin.js#L100)

(there's also `projects/plugins/jetpack/modules/carousel/swiper-bundle.js` and `projects/plugins/jetpack/modules/shortcodes/js/jmpress.js`, but those seem like copy-pasted external libraries so I skipped them)

</details>

Removing all the uses should also allow us to drop the dep on `@wordpress/keycodes` from [Jetpack's package.json](https://github.com/Automattic/jetpack/blob/add/remove-deprecated-event-keycode/projects/plugins/jetpack/package.json#L137).

**Comment by sharpirate** [comment]
Thanks for the input @anomiex! Using `event.code` is a great point, I forgot that was event a thing. I'll go ahead and update the PR. The reason I changed only a few places was because this is my first PR to jetpack as part of my onboarding and wanted to keep the changes minimal to avoid breaking something. Let me know if it makes more sense to refactor this everywhere (the files you mentioned)

**Comment by anomiex** [comment]
It's fine to only to a part of the code if that's the part you're working on, or to divide the refactoring into pieces to make them easier to test. I only asked because the title and description here made it sound like fixing everything was intended.

**Review by anomiex — APPROVED** [review]

**Pull request merged**